contracts/examples/dao/DAO.sol

```diff
@@ -49,83 +57,61 @@ contract DAO is VentureEth {
+            address(this),
+            abi.encodeWithSignature("fundVenture(address,uint256)", venture, funding)
         );
-        totalVotesByHolder[msg.sender] = totalVotesByHolder[msg.sender].add(
-            votes
+        voting.registerProposal(
+            address(this),
+            abi.encodeWithSignature("retrieveVentureTokens(address)", venture)
         );
-        votesForVentureByHolder[venture][msg.sender] = votesForVentureByHolder[
-            venture][msg.sender].add(votes);
-        addBackerForVenture(venture, msg.sender);
+        voting.open();
+        proposals[venture] = address(voting);
+        ventures.add(venture);
+        emit VentureProposed(venture, address(voting));
     }
 
     /**
-     * @notice Fund venture. Can only be used after the original funding round, and after voting total is over 50% + 1 of total DAO tokens.
-     * @param venture The venture to fund. Must be of type VentureEth.
+     * @notice Fund a venture proposal.
+     * @param venture The address of the VentureEth contract to fund.
+     * @param funding The ether to fund the venture with.
      */
-    function fundVenture(address venture) public {
-        require(currentState == "LIVE", "DAO needs to be LIVE.");
-        require(
-            totalVotesForVenture[venture] >= this.totalSupply().div(2).add(1),
-            "Not enough expressed votes."
-        );
-        uint256 amount = proposedFundingForVenture[venture];
-        fundingPool = fundingPool.sub(amount);
-        address backer;
-        for (uint256 i = 0; i < backersForVenture[venture].length; i++) {
-            backer = backersForVenture[venture][i];
-            totalVotesByHolder[backer] = totalVotesByHolder[backer].sub(
-                votesForVentureByHolder[venture][backer]
-            );
-        }
-        delete totalVotesForVenture[venture];
-        delete proposedFundingForVenture[venture];
-        // solium-disable-next-line security/no-call-value
-        VentureEth(venture).invest.value(amount)();
+    function fundVenture(
+        address venture,
+        uint256 funding
+    ) public {
+        require(proposals[venture] == msg.sender);
+        VentureEth(venture).invest
+            .value(funding)();  // Maybe use ERC165 to make sure it's a VentureEth
     }
 
     /**
-     * @notice Claims issuance tokens from funded venture. Can only be used after the original funding round, and after funding the venture.
-     * @param venture The venture to claim tokens from. Must be of type VentureEth.
+     * @notice Retrieve tokens minted for the DAO after an investment.
+     * @param venture The address of the VentureEth contract to retrieve tokens from.
      */
-    function claimTokensForFundedVenture(address venture) public {
-        require(currentState == "LIVE", "DAO needs to be LIVE.");
+    function retrieveVentureTokens(
+        address venture
+    ) public {
+        require(proposals[venture] == msg.sender);
```

Comment:
We don't need this `require`, actually. Anyone can make the DAO retrieve venture tokens.